src/plugins/vis_type_table/public/components/table_vis_component.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useCallback, useMemo } from 'react';
+import { EuiDataGridProps, EuiDataGrid } from '@elastic/eui';
+
+import { IInterpreterRenderHandlers } from 'src/plugins/expressions';
+import { Table } from '../table_vis_response_handler';
+import { TableVisConfig } from '../types';
+
+interface TableVisComponentProps {
+  table: Table;
+  visConfig: TableVisConfig;
+  handlers: IInterpreterRenderHandlers;
+}
+
+export const TableVisComponent = ({ table, visConfig, handlers }: TableVisComponentProps) => {
+  const { rows, columns } = table;
+  const [pagination, setPagination] = useState({ pageIndex: 0, pageSize: 5 });
+  const onChangeItemsPerPage = useCallback(
+    (pageSize) => setPagination((p) => ({ ...p, pageSize, pageIndex: 0 })),
```

Comment:
Is it necessary to reset the index? It'd be a better UX to recalculate the page index so the user can keep their place while changing the page size.

Reply:
Yes. the whole pagination feature will be re-write in the following PR. This is just for hook up and see a rough table vis purpose.